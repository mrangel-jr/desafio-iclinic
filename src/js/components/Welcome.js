/* eslint-disable react/prop-types */
import React, { useContext } from "react";
import { useHistory } from "react-router-dom";
import {MainContext} from './MainContext'

export default function Welcome() {
	const history = useHistory();
	const {fetchData} = useContext(MainContext);

	function handleClick() {
		fetchData().then(() => history.push("/character"))
	}

	return (
		<div className="welcome">
			<div className="welcome_iclinic--title">
				Welcome to <b>iClinic</b>
			</div>
			<p className="welcome_iclinic--subtitle">FRONTEND CHALLENGE</p>
			<button onClick={handleClick} className="welcome_iclinic--start">
				START
			</button>
		</div>
	);
}
