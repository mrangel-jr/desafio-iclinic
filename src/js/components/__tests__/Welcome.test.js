import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import Welcome from "../Welcome";

describe("Welcome", () => {
	test("snapshot renders", () => {
		const component = renderer.create(<Welcome />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("Check if show button", () => {
		const wrapper = mount(<Welcome />);
		const button = wrapper.find("button");

		expect(button.length).toEqual(1);
	});
});
