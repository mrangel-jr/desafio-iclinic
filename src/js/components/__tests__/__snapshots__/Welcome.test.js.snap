// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome snapshot renders 1`] = `
<div
  className="welcome"
>
  <div
    className="welcome_iclinic--title"
  >
    Welcome to 
    <b>
      iClinic
    </b>
  </div>
  <p
    className="welcome_iclinic--subtitle"
  >
    FRONTEND CHALLENGE
  </p>
  <button
    className="welcome_iclinic--start"
    onClick={[Function]}
  >
    START
  </button>
</div>
`;
